service/kernel/oneapi/build_program.h

```diff
@@ -0,0 +1,61 @@
+/* file: build_program.h */
+/*******************************************************************************
+* Copyright 2015-2019 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+/*
+//++
+//  Declaration of build_program
+//--
+*/
+
+#ifndef __BUILD_PROGRAM_H__
+#define __BUILD_PROGRAM_H__
+
+#include "math_service_types.h"
+#include "services/buffer.h"
+#include "cl_kernels/sum_reducer.cl"
+#include "service_defines_oneapi.h"
+#include "oneapi/internal/types_utils.h"
+#include "oneapi/internal/execution_context.h"
+
+namespace daal
+{
+namespace oneapi
+{
+namespace internal
+{
+
+static void buildProgram(ClKernelFactoryIface& kernelFactory,
```

Comment:
If you want us to use this function to build all the kernels inside the library, please, do such changes not only for `SumReducer`